domain/src/main/java/org/oppia/domain/audio/CellularDialogController.kt

```diff
@@ -0,0 +1,46 @@
+package org.oppia.domain.audio
+
+import androidx.lifecycle.LiveData
+import org.oppia.app.model.CellularDataPreference
+import org.oppia.data.persistence.PersistentCacheStore
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.data.DataProviders
+import org.oppia.util.logging.Logger
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Controller for persisting and retrieving the cellular data preference. */
+@Singleton
+class CellularDialogController @Inject constructor(
+  cacheStoreFactory: PersistentCacheStore.Factory, private val dataProviders: DataProviders,
+  private val logger: Logger
+) {
+  private val cellularDataStore = cacheStoreFactory.create("cellular_data_preference", CellularDataPreference.getDefaultInstance())
+
+  /** Saves that the user's preference on whether to hide the dialog. */
+  fun setHideDialogPreference(hideDialog: Boolean) {
+    cellularDataStore.storeDataAsync(updateInMemoryCache = true) {
+      it.toBuilder().setHideDialog(hideDialog).build()
+    }.invokeOnCompletion {
+      it?.let {
+        logger.e("DOMAIN", "Failed when storing the user's preference to hide cellular data dialog.", it)
+      }
+    }
+  }
+
+  /** Saves that the user's preference on whether to use cellular data. */
+  fun setUseCellularDataPreference(useData: Boolean) {
```

Comment:
Could this be combined with the above? I suspect we only ever want to save the user's preference if they don't want to show the dialog again. Perhaps:

``setNeverUseCellularDataPreference()``
``setAlwaysUseCellularDataPreference()``

(these would call a common private helper that always sets the hide dialog bit to true and sets the corresponding cellular preference). Longer term we could have a third method to reset the data source if we add a preference for this in the user settings menu.